es6-persistence/src/main/java/com/netflix/conductor/dao/es6/index/ElasticSearchDAOV6.java

```diff
@@ -222,12 +211,11 @@ private void shutdownExecutorService(ExecutorService execService) {
     public void setup() throws Exception {
         waitForHealthyCluster();
 
-        createIndexesTemplates();
-
-        createWorkflowIndex();
-
-        createTaskIndex();
-
+        if (config.isElasticSearchAutoIndexManagementEnabled()) {
```

Comment:
(Following up on our conversation)
I think this could be refactored to use an in-memory queue and an indexer class when index management is enabled. That way we'll have comparable architecture in both cases (producer -> queue -> indexer)

All this logic would remain the same, but isolated in it's own class, and that class can be re-used from the indexer service, when you decide your deployment requires an independent service.